src/utils/work/seekWorkResponse.js

```diff
@@ -0,0 +1,101 @@
+import moment from 'moment';
+import Amplify, { Storage } from 'aws-amplify';
+
+import connectionPromise from '../socketConnection';
+import WorkResponseError from '../WorkResponseError';
+import getAuthJWT from '../getAuthJWT';
+import WorkTimeoutError from '../WorkTimeoutError';
+
+const seekFromS3 = async (ETag) => {
+  const configuredBucket = Amplify.configure().Storage.AWSS3.bucket;
+  const storageUrl = await Storage.get(
+    ETag,
+    {
+      bucket: configuredBucket.replace('biomage-originals', 'worker-results'),
+    },
+  );
+
+  const storageResp = await fetch(storageUrl);
+  if (!storageResp.ok) {
+    return null;
+  }
+
+  const response = await storageResp.json();
+
+  return response;
+};
+
+const seekFromAPI = async (
+  experimentId,
+  body,
+  timeout,
+  ETag,
+  eventCallback,
```

Comment:
what is the purpose of the `eventCallback`?

Reply:
See above.